compiler/semtypes.nim

```diff
@@ -456,8 +457,11 @@ proc semCaseBranch(c: PContext, t, branch: PNode, branchIndex: int,
     else:
       # constant sets and arrays are allowed:
       var r = semConstExpr(c, b)
-      # for ``{}`` we want to trigger the type mismatch in ``fitNode``:
-      if r.kind notin {nkCurly, nkBracket} or len(r) == 0:
+      if r.kind in {nkCurly, nkBracket} and len(r) == 0  and sonsLen(branch)==2:
```

Comment:
I thought they are redundant too at first. But if I move that in there it would still need to check sonsLen(branch)==2 to only accept it when it is a single `{}`. Which would introduce a second error case. I don't see how to optimize this without changing the logic (allowing some strange variants of mixed cases) or introducing another block with replicated code.